java-frontend/src/main/java/org/sonar/plugins/java/api/tree/PatternInstanceOfTree.java

```diff
@@ -39,6 +39,13 @@ public interface PatternInstanceOfTree extends ExpressionTree {
 
   SyntaxToken instanceofKeyword();
 
+  @Deprecated
+  /**
+   * @deprecated Use {@link PatternInstanceOfTree#pattern()}
+   * @since 7.16
+   */
```

Comment:
I would place the `@Deprecated` annotation after the javadoc (convention).
Also, I would add the `@Nullable` annotation, since when the pattern won't always be TypePattern, you might not always have variables inside.